frontend/src/concepts/k8s/NameDescriptionField.tsx

```diff
@@ -24,75 +38,110 @@ const NameDescriptionField: React.FC<NameDescriptionFieldProps> = ({
-  }, [showK8sName, data.name]);
+      setData({
+        ...data,
+        name,
+        k8sName: { value: truncatedk8sName, isTruncated: true, isUserInputk8sName: false },
+      });
+    }
+    if (name.length === 0 || data.k8sName?.value.length === 0) {
+      setValidateK8sName({
+        ruleCharacterTypes: 'indeterminate',
+        ruleLength: 'indeterminate',
+      });
+    }
+  };
 
   return (
     <Stack hasGutter>
       <StackItem>
-        <FormGroup label="Name" isRequired fieldId={nameFieldId}>
+        <FormGroup
+          label="Name"
+          isRequired
+          fieldId={nameFieldId}
+          labelInfo={
+            <ToggleResourceName
+              showHiddenResourceName={showHiddenResourceName}
+              setShowHiddenResourceName={setShowHiddenResourceName}
+            />
+          }
+        >
           <TextInput
             isRequired
             ref={autoSelectNameRef}
             id={nameFieldId}
             name={nameFieldId}
             value={data.name}
-            onChange={(name) => setData({ ...data, name })}
+            onChange={handleNameChange}
           />
         </FormGroup>
       </StackItem>
-      {showK8sName && (
-        <StackItem>
-          <FormGroup
-            label="Resource name"
-            labelIcon={
-              <Tooltip
-                removeFindDomNode
-                position="right"
-                content={
-                  <Stack hasGutter>
-                    <StackItem>
-                      Resource names are what your resources are labeled in OpenShift.
-                    </StackItem>
-                    <StackItem>Resource names are not editable after creation.</StackItem>
-                  </Stack>
-                }
-              >
-                <HelpIcon aria-label="More info" />
-              </Tooltip>
-            }
-            isRequired
-            fieldId={`resource-${nameFieldId}`}
-            helperText={
-              !disableK8sName &&
-              "Must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character"
-            }
-          >
-            <TextInput
-              isRequired
-              isDisabled={disableK8sName}
-              id={`resource-${nameFieldId}`}
-              name={`resource-${nameFieldId}`}
-              value={data.k8sName ?? k8sName}
-              onChange={(k8sName) => {
-                setData({ ...data, k8sName });
+      {showHiddenResourceName && (
+        <FormFieldGroup
+          header={
+            <FormFieldGroupHeader
+              titleText={{
+                text: <HeadingAdvancedSettings />,
+                id: 'field-group-advanced-settings',
               }}
-              validated={!isValidK8sName(data.k8sName) ? 'error' : undefined}
+              titleDescription="Resource name usually use the Name as the value. You can view the Resource name and edit the Resource name during creation process."
```

Comment:
I didn't see this here in the [mocks](https://www.figma.com/proto/9CkrqD7byqvTBUzRok2ihE/RHODS?page-id=1334%3A31100&type=design&node-id=1334-35808&viewport=642%2C570%2C0.17&t=0WZlbQ8nJk6mYqyJ-1&scaling=scale-down-width&starting-point-node-id=1334%3A35808), let me know if I'm looking at some outdated mocks.

Reply:
@manaswinidas Do you mean [this page](https://www.figma.com/proto/9CkrqD7byqvTBUzRok2ihE/RHODS?page-id=1334%3A31100&type=design&node-id=1334-47293&viewport=642%2C570%2C0.17&t=0WZlbQ8nJk6mYqyJ-1&scaling=scale-down-width&starting-point-node-id=1334%3A35808)? There is a popover for the Resource name.

Reply:
@uidoyen can you update according to the mocks provided in the previous comment?

Reply:
@manaswinidas It is based on this comment here https://github.com/opendatahub-io/odh-dashboard/issues/1627#issuecomment-1751941678